### DISCUSSION TITLE
Default data for new one to one table

### DISCUSSION THREAD

**Discussion by wangcode** [prisma/prisma#22726] [Q&A] [answered]
### Question

If I already have an user table and some data，then i create a new table one to one relation to user table, but I want each user have record in this table, If all fields have default values except userid, how can i seed default data for this table using prisma? It’s migration job? What is the right way?
Sorry I’m new to database.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @wangcode 

Thank you for raising this question.

Seeding a database with Prisma ORM typically involves creating a seed script that populates the database with the necessary data. This is not directly a part of the migration process, which is primarily concerned with creating and updating the database schema. Seeding is a separate step that you can run after the migration to populate the new tables with data.

Since you want to create a one-to-one relationship and ensure that each user has a corresponding record in the new table, you would need to write a seed script that iterates over each user in the existing User table and creates a new record in the related table with the default values and the userId. To seed a one-to-one relationship table where each user must have a corresponding record, you would typically:

1. Retrieve all users from the User table.
2. Iterate over each user and create a related record in the new table with the userId and any default values for other fields.
The code for such an operation would look something like this:
```js
async function seedOneToOneRelation() {
  // Retrieve all users
  const users = await prisma.user.findMany();

  // Iterate over each user and create a related record in the new table
  for (const user of users) {
    await prisma.newTable.create({
      data: {
        userId: user.id,
        // Set other fields with default values
        otherField: 'defaultValue',
        // ...
      },
    });
  }

  console.log('Seeding one-to-one relation table completed.');
}

seedOneToOneRelation()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by wangcode** [reply]
Thank you for your patient answer, thank you very much.
